verify/cert.go

```diff
@@ -19,17 +19,26 @@ import (
 	"golang.org/x/exp/slices"
 )
 
-// CertVerifier verifies the DA certificate against on-chain EigenDA contracts
+type CertVerifier interface {
+	verifyBatchConfirmedOnChain(ctx context.Context, batchID uint32, batchMetadata *disperser.BatchMetadata) error
+	verifyMerkleProof(inclusionProof []byte, root []byte, index uint32, blobHeader BlobHeader) error
+	quorumNumbersRequired(opts *bind.CallOpts) ([]byte, error)
+	quorumAdversaryThresholdPercentages(opts *bind.CallOpts) ([]byte, error)
+}
+
+// CertVerification verifies the DA certificate against on-chain EigenDA contracts
 // to ensure disperser returned fields haven't been tampered with
-type CertVerifier struct {
+type CertVerification struct {
```

Comment:
knit - any  reason to make this public if we're binding to an interface?

Reply:
Not really, I can change. Don't think using an interface matters here though, it's more about whether we think people will want to import and use this struct elsewhere.

Reply:
[bc6d810](https://github.com/Layr-Labs/eigenda-proxy/pull/226/commits/bc6d8109fd710c91379a13e331c2898a4524e2b4)